teps/0130-pipeline-level-docker-daemon.md

```diff
@@ -0,0 +1,409 @@
+
+### Simplicity
+
+The current user experience isn't flexible enough for many common docker-related use cases involving more complex logic than a build and push.
+
+### Flexibility
+
+Allowing users to specify Pipeline-level sidecars is more flexible than first class support for building images,
+like Tekton previously had with the Image PipelineResource.
+It's an open question whether we'd like to allow more flexibility in terms of other resources (such as custom Tasks)
+with the same lifespan as PipelineRuns.
+
+### Conformance
+
+Sidecar configuration already contains some Kubernetes-specific configuration,
+such as liveness/readiness probes. This proposal doesn't introduce new Kubernetes-specific
+configuration, but it allows it to be specified in Pipelines as well as Tasks.
+
+### User Experience
+
+<!--
+(optional)
+
+Consideration about the user experience. Depending on the area of change,
+users may be Task and Pipeline editors, they may trigger TaskRuns and
+PipelineRuns or they may be responsible for monitoring the execution of runs,
+via CLI, dashboard or a monitoring system.
+
+Consider including folks that also work on CLI and dashboard.
+-->
+
+### Performance
+
+<!--
+(optional)
+
+Consider which use cases are impacted by this change and what are their
+performance requirements.
+- What impact does this change have on the start-up time and execution time
+of TaskRuns and PipelineRuns?
+- What impact does it have on the resource footprint of Tekton controllers
+as well as TaskRuns and PipelineRuns?
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Tekton ecosystem. Consider including folks that also work outside the WGs
+or subproject.
+- How will security be reviewed and by whom?
+- How will UX be reviewed and by whom?
+-->
+
+### Drawbacks
+
+<!--
+Why should this TEP _not_ be implemented?
+-->
+
+- If docker TLS certs are used, they must now be bound in a persistent volume claim to share
+between multiple Tasks, rather than an emptyDir volume to share between multiple steps.
+This is a pretty heavyweight option for a relatively small amount of data.
+However, this is more an artifact of the limited options Pipelines currently has for sharing
+data between Tasks, which should be treated as a separate problem.
+
+## Alternatives
+
+<!--
+What other approaches did you consider and why did you rule them out? These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+Possible alternatives:
+- User writes a Task at the beginning of the PipelineRun to start up a sidecar service and a Finally Task to tear it down
+  (i.e. we do nothing except maybe improve our documentation)
+- First-class docker support in the API; e.g. a user can specify that they'd like their Pipeline to have a docker daemon provided
```

Comment:
This would be a *no no no* from me. I see no reason to bubble any "tooling specific" thing to the API. If we do that for docker, what prevent us from doing that from any other "daemon" out there (builtkitd, …).